From: Rene Kita <mail@rkta.de>
Subject: Simplify reshapeBuffer

Every single place where we call reshapeBuffer we set
buffer->need_reshape to TRUE. In reshapeBuffer we return early if
buffer->need_reshape is FALSE - which cannot happen - and set it to
FALSE when we continue.

Remove the check inside reshapeBuffer. If we call this function, we mean
it.

---
 buffer.c  |    2 --
 display.c |    1 -
 main.c    |    3 ---
 3 files changed, 6 deletions(-)

--- a/buffer.c
+++ b/buffer.c
@@ -473,8 +473,6 @@ reshapeBuffer(Buffer *buf)
     wc_uint8 old_auto_detect = WcOption.auto_detect;
 #endif
 
-    if (!buf->need_reshape)
-	return;
     buf->need_reshape = FALSE;
     buf->width = INIT_BUFFER_WIDTH;
     if (buf->sourcefile == NULL)
--- a/display.c
+++ b/display.c
@@ -385,7 +385,6 @@ displayBuffer(Buffer *buf, int mode)
 	 (is_html_type(buf->type) || FoldLine || !strcmp(buf->type,
 							 "text/gemini")))
 	|| buf->need_reshape) {
-	buf->need_reshape = TRUE;
 	reshapeBuffer(buf);
     }
     if (showLineNum) {
--- a/main.c
+++ b/main.c
@@ -4921,7 +4921,6 @@ DEFUN(vwSrc, SOURCE VIEW, "Toggle betwee
     buf->clone = Currentbuf->clone;
     (*buf->clone)++;
 
-    buf->need_reshape = TRUE;
     reshapeBuffer(buf);
     pushBuffer(buf);
     displayBuffer(Currentbuf, B_NORMAL);
@@ -5012,7 +5011,6 @@ DEFUN(geminize, GEMINIZE, "Toggle betwee
     buf->clone = Currentbuf->clone;
     (*buf->clone)++;
 
-    buf->need_reshape = TRUE;
     reshapeBuffer(buf);
     pushBuffer(buf);
     displayBuffer(Currentbuf, B_NORMAL);
@@ -5147,7 +5145,6 @@ DEFUN(reload, RELOAD, "Load current docu
 /* reshape */
 DEFUN(reshape, RESHAPE, "Re-render document")
 {
-    Currentbuf->need_reshape = TRUE;
     reshapeBuffer(Currentbuf);
     displayBuffer(Currentbuf, B_FORCE_REDRAW);
 }
